src/ir/ir_visitor.h

```diff
@@ -24,13 +24,13 @@ class Operation;
 
 // An interface for the visitor class. We will also pass in the `Derived` class
 // as template argument if we want to support CRTP at a later point.
-template <typename Derived>
+template <typename Derived, typename Result=void>
```

Comment:
Let us keep this consistent with `IRTraversingVisitor` and make it `Result=Unit`?

Reply:
Done (surprised that didn't break anything :D )